src/GogeToken.sol

```diff
@@ -596,346 +584,350 @@ contract DogeGaySon is ERC20, Ownable {
-            if (_address == dexList[s]) return (true, s);
+    function isExcludedFromCirculatingSupply(address _address) public view returns(bool, uint8) {
+        for (uint8 i = 0; i < excludedFromCirculatingSupply.length; i++){
+            if (_address == excludedFromCirculatingSupply[i]) {
+                return (true, i);
+            }
         }
         return (false, 0);
     }
 
-    function setAutomatedMarketMakerPair(address pair, bool value) public {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(pair != uniswapV2Pair, "The PancakeSwap pair cannot be removed from automatedMarketMakerPairs");
-        _setAutomatedMarketMakerPair(pair, value);
-    }
+    function excludeFromCirculatingSupply(address account, bool excluded) public {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::excludeFromCirculatingSupply() not authorized");
 
-    function _setAutomatedMarketMakerPair(address pair, bool value) private {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(automatedMarketMakerPairs[pair] != value, "Automated market maker pair is already set to that value");
-        automatedMarketMakerPairs[pair] = value;
+        (bool _isExcluded, uint8 i) = isExcludedFromCirculatingSupply(account);
+        require(_isExcluded != excluded, "GogeToken.sol::excludeFromCirculatingSupply() account already set to that boolean value");
 
-        if(value) {
-            cakeDividendTracker.excludeFromDividends(pair);
-            (bool _isDex, ) = isDex(pair);
-            if(!_isDex) dexList.push(pair);        
+        if(excluded) {
+            if (!cakeDividendTracker.excludedFromDividends(account)) {
+                cakeDividendTracker.excludeFromDividends(account);
+            }
+            if(!_isExcluded) excludedFromCirculatingSupply.push(account);        
         } else {
-            (bool _isDex, uint8 s) = isDex(pair);
-            if(_isDex){
-                dexList[s] = dexList[dexList.length - 1];
-                dexList.pop();
+            if(_isExcluded){
+                excludedFromCirculatingSupply[i] = excludedFromCirculatingSupply[excludedFromCirculatingSupply.length - 1];
+                excludedFromCirculatingSupply.pop();
             } 
         }
 
+        emit AddressExcludedFromCirculatingSupply(account, excluded);
+    }
+
+    function setAutomatedMarketMakerPair(address pair, bool value) external {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::setAutomatedMarketMakerPair() not authorized");
+        require(pair != uniswapV2Pair, "GogeToken.sol::setAutomatedMarketMakerPair() the PancakeSwap pair cannot be removed from automatedMarketMakerPairs");
+        _setAutomatedMarketMakerPair(pair, value);
+    }
+
+    function _setAutomatedMarketMakerPair(address pair, bool value) internal {
```

Comment:
Still feel like this could just be made into one function.

Reply:
The internal function is multi-purpose similar to the _mint function. If called externally it should only be called by the owner address which is why there's a separate setAutomatedMarketMakerPair() for external use. But _setAutomated...() could be called internally to serve the same purpose for example in the constructor and if I ever need to exclude an address or pair from circulating supply.